src/aind_behavior_curriculum/behavior.py

```diff
@@ -0,0 +1,45 @@
+"""
+Base Behavior Models
+"""
+
+from pydantic import BaseModel, ConfigDict, Field
+
+
+class AindBehaviorModel(BaseModel):
+    """
+    Defines Pydantic configurations applied to all behavior models.
+    BaseModel: Validate arguments on initialization.
+    Configurations:
+        - extra='forbid':
+            Do not allow a model to be initalized with undocumented parameters.
+        - validate_assignment=True:
+            Revalidate fields against schema on any change to model instance.
+        - strict=True:
+            Enforce strict typing.
+    """
+
+    model_config = ConfigDict(
+        extra="forbid", validate_assignment=True, strict=True
```

Comment:
Having extra="forbid" will prevent deserialization of the child json back to the parent, which was one of the things I mentioned we be nice to preserve.

Consider: 

```python
from aind_behavior_curriculum.behavior import Task
from pydantic import Field


class MyTask(Task):
    prop1: int = Field(1, description="This is a property")
    prop2: str = Field("a", description="This is another property")


instance = MyTask(name="Task")
print(instance)
# Task(name='Task', description='', prop1=1, prop2='a')

instance_json = instance.model_dump_json()
print(instance_json)
#{"name":"Task","description":"","prop1":1,"prop2":"a"}

#instance_parent = Task.model_validate_json(instance_json)
# throws error due to extra inputs
```

The alternative of having the model `extra` as `ignore` or `allow` might not be the best either. `ignore` will drop the child values as they are not known to the parent. This is not that bad but it is a pitty since extra information wouldn't hurt (this would be the best low hanging fruit choice). `allow` will make things too messy and error-prone, so i would just not use it at all. I think the best solution would be to implement a property in the base class task, called `task_parameters` that is modeled how Jon and I suggested  in this PR https://github.com/AllenNeuralDynamics/aind-data-schema/pull/736#issue-2103221565